### DISCUSSION TITLE
How can All Users be fetched With tortoise ORM?

### DISCUSSION THREAD

**Discussion by publio-cornelio** [fastapi-users/fastapi-users#910] [Q&A] [answered]
First of all I would like to thank the producers of this module, I find it, in my limited experience, very useful.

Unfortunately, I'm having some trouble getting a view of all users with a request. I found an example (How can I get all users? #520) but it applied to MongoDB and I am using Tortoise ORM. Applying the user creation exercise inside the documentation, I manage to have a database contextmanager inside a route, the database instance appears as models.UserModel, 

     async with get_user_db_context() as user_db:
        
                print(user_db.model)                         =>  <class 'models.UserModel'>
     
                print(dir(user_db)) != print(dir(UserModel))        => don't match

but when I evaluate the disposable methods (with a dir()), the are different between the tortoise.UserModel and the instance of the database, and I can,t find a method .all. I don't have much experience in Python and I can't move forward. Therefore, I would appreciate if someone could offer an example on how to get all Users for the Tortoise ORM. I think it would be useful for understanding better the program.

Sincerely.

**Comment by publio-cornelio** [comment]
I understand that the purpose of this forum is not that one responds to oneself, but after detailing the problem in writing, I made  my mind up with the solution.
As I have said I have got user_db through the get_user_db_context. I was wrong, user_db doesn't return a UserModel, but user_db.model() does. So, we can use it adding .all().
The result feeds a Pydantic class, and the only present in models is UsersDB.
Two lines, but pretty hard to find.

      @app.get("/protected-route")
      async def protected_route ():
       
            async with get_user_db_context() as user_db:
       
                 return await UserDB.from_queryset(user_db.model().all())

**Comment by frankie567** [comment, marked as answer]
The solution you show can work indeed, but there is actually a much easier way.

In the case you're presenting here, you're out of the context of FastAPI Users, so you actually don't need to use the tools provided: you can directly use Tortoise ORM methods, like for any other models in your application:

```py
@app.get("/users")
async def list_users():
    users = await UserModel.all()
    return UserDB.from_queryset(users)
```

**Reply by publio-cornelio** [reply]
Thanks. Merci bien.
